src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -915,13 +919,37 @@ extern "C" int32_t SystemNative_GetDomainName(uint8_t* name, int32_t nameLength)
     assert(name != nullptr);
     assert(nameLength > 0);
 
+#if HAVE_GETDOMAINNAME
 #if HAVE_GETDOMAINNAME_SIZET
     size_t namelen = UnsignedCast(nameLength);
 #else
     int namelen = nameLength;
 #endif
 
     return getdomainname(reinterpret_cast<char*>(name), namelen);
+#elif HAVE_UNAME
+    // On Android, there's no getdomainname but we can use uname to fetch the domain name
+    // of the current device
+    size_t namelen = UnsignedCast(nameLength);
+    utsname  uts;
+
+    // If uname returns an error, bail out.
+    if(uname(&uts) == -1) {
+        return -1;
+    }
+
+    // If we don't have enough space to copy the name, bail out.
+    if(strlen(uts.domainname) >= namelen) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    // Copy the domain name
+    strncpy(reinterpret_cast<char*>(name), uts.domainname, static_cast<size_t>(namelen));
```

Comment:
Could you change this to use the SafeStringCopy helper we have here?
https://github.com/dotnet/corefx/blob/master/src/Native/Unix/Common/pal_utilities.h#L106